daemon/log.c

```diff
@@ -332,14 +292,11 @@ void log_bin(MTC_S32 priority, PMTC_S8 data, MTC_S32 size)
         return;
     }
 
-#if USE_SEMAPHORE
-    sem_wait(semaphore);
-#else
-    pthread_spin_lock(&lock);
-#endif
+    pthread_rwlock_rdlock(&lock);
```

Comment:
Similar comment to before that we don't need to hold this unless we're actually going to log to the file etc